projects/gateway2/setup/ggv2setup_test.go

```diff
@@ -240,14 +241,41 @@ func TestScenarios(t *testing.T) {
 				// that we get test pollution.
 				// once we change it to only include the ones in the proxy, we can re-enable this
 				//				t.Parallel()
-				testScenario(t, ctx, client, xdsPort, fullpath)
+				testScenario(t, ctx, setupOpts.KrtDebugger, snapCache, client, xdsPort, fullpath)
 
 			})
 		}
 	}
 }
 
-func testScenario(t *testing.T, ctx context.Context, client istiokube.CLIClient, xdsPort int, f string) {
+func dumpCache(snapCache cache.SnapshotCache) iosnapshot.SnapshotResponseData {
+	cacheKeys := snapCache.GetStatusKeys()
+	cacheEntries := make(map[string]interface{}, len(cacheKeys))
+
+	for _, k := range cacheKeys {
+		xdsSnapshot, err := getXdsSnapshot(snapCache, k)
+		if err != nil {
+			cacheEntries[k] = err.Error()
+		} else {
+			cacheEntries[k] = xdsSnapshot
+		}
+	}
+	return iosnapshot.SnapshotResponseData{
+		Data:  cacheEntries,
+		Error: nil,
+	}
+
+}
+func getXdsSnapshot(snapCache cache.SnapshotCache, k string) (cache cache.Snapshot, err error) {
```

Comment:
Should this live in https://github.com/solo-io/gloo/tree/main/projects/gloo/pkg/xds? If getting the snapshot from the cache can panic, should this functionality also be used in our control plane directly?

Reply:
it is used there - see it in `history.go